matrix_sdk/src/http_client.rs

```diff
@@ -125,10 +122,6 @@ impl HttpClient {
             None => self.request_config,
         };
 
-        #[cfg(not(feature = "appservice"))]
-        let request = self.try_into_http_request(request, session, config).await?;
-
-        #[cfg(feature = "appservice")]
         let request = if !self.request_config.assert_identity {
             self.try_into_http_request(request, session, config).await?
         } else {
```

Comment:
Given that `assert_identity` can only switched with `appservice` feature active this should be OK as runtime check